### DISCUSSION TITLE
Safe Area Aware Example On iOS and Android

### DISCUSSION THREAD

**Discussion by Joshua-Beatty** [excaliburjs/Excalibur#2349] [Show and tell]
I've been having a lot of trouble getting safe area insets in a single way across different platforms and then using those insets within Excalibur, so I created a [simple capacitor plugin](https://www.npmjs.com/package/get-safe-area-capacitor) to just call the native code in order to retrieve the insets. I am not fully confident in the inner workings of all of capacitor at the moment so use this at your own risk, or maybe just take a look at the plugin code for some ideas.

Here is a simple demo I have put together using capacitor and Excalibur:  ([Repository here](https://github.com/Joshua-Beatty/capacitor-excalibur-safe-area-example))

It is just a simple demo where you can click to add pink squares, and there is a grey bar off to the left that will always stay within the intractable portion of the screen.

iOS:
![image](https://user-images.githubusercontent.com/13110124/173241824-fe5055e8-9b12-4024-97d3-e0b577afdc62.png)

Android:
![image](https://user-images.githubusercontent.com/13110124/173241836-33f17341-d2fe-48c8-8ce6-99c201588ccf.png)

Web:
![image](https://user-images.githubusercontent.com/13110124/173241844-44eee8e4-d972-4051-b667-616114c9c5f0.png)

Currently the grey bars position is just statically calculated based off the assumption that the center of the screen is also the center of the world, so for moving cameras and other things along those lines the code would probably need to be adjusted.

Hopefully this helps some people or maybe can share some ideas around the community about dealing with safe areas.

**Comment by eonarheim** [comment]
@Joshua-Beatty thanks for sharing! 

This is definitely something I'll be using in my next mobile game!

Do you mind if I shout this plugin out in a video soon(ish)?

**Reply by Joshua-Beatty** [reply]
Not at all, Feel Free!
